src/detect-ipproto.c

```diff
@@ -121,20 +122,23 @@ static DetectIPProtoData *DetectIPProtoParse(const char *optstr)
 
     /* Protocol name/number */
     if (!isdigit((unsigned char)*(args[1]))) {
-        struct protoent *pent = getprotobyname(args[1]);
-        if (pent == NULL) {
-            SCLogError(SC_ERR_INVALID_VALUE, "Malformed protocol name: %s",
-                       str_ptr);
+        int proto = SCGetProtoByName(args[1]);
+        if (-1 == proto) {
+            SCLogError(SC_ERR_INVALID_VALUE, "Malformed protocol name: \"%s\"", str_ptr);
```

Comment:
Is 'Malformed' the correct term here? Could it also just be unknown

Reply:
Will update.